### DISCUSSION TITLE
Display weather forecast images on SH1106

### DISCUSSION THREAD

**Discussion by ivanvach** [arendst/Tasmota#21148] [Q&A]
Few questions here

1. Are there stock images for weather forecast that I can use in Tasmota such as SUNNY, CLOUDY, RAINY, STORMY
2. If YES than how to show such images in the display conditionally using the pressure readings from my sensor that are available in Tasmota
"BME280":{"Temperature":76.0,"Humidity":41.0,"DewPoint":50.5,"Pressure":1003.3},"PressureUnit":"hPa","TempUnit":"F"}

If pressure <990 STORMY
If pressure <1000 RAINY
If pressure <1004 CLOUDY
If pressure >1004 SUNNY

3. If NO than how to resize, store and use external images that I find online.

Thanks

**Comment by valdus2** [comment]
1) I don't think there are stock images in Tasmota but you can use whatever image you like. Also I don't think you can display graphics in an oled display.

2) Your approach is so much simplistic and that is not the way for what the weather is or what the the weather will be...
Knowing just the number itself, isn’t all that helpful. What matters is in which direction the pressure is moving. It’s the change in barometric pressure that helps forecast the weather. This means that static numbers with no indication of rising or falling aren’t generally very useful. 
So, to get accurate predictions, you need to take a look at the barometer every few minutes-hours and see how the number is changing. 
So what you want to know is if it's rising or falling, the trend. This will be more helpful and accurate for an amateur meteorologist
Also keep in mind that barometers will only help you forecast for the next 24-48 hours or so. 
Since pressure systems last days and sometimes weeks, it can be a while before you notice any substantial difference in your readings.

For your exact question, using rules for example:

on BME280#Pressure<990 do DisplayText[P/STORMY.rgb:] endon

3) 
You use the command :
Pfilename: = display an rgb 16-bit color (or jpg on ESP32) image when file system is present, 

[https://tasmota.github.io/docs/Displays/#miscellaneous](url)

Scripteditor contains a converter to convert jpg to special RGB16 pictures See ScriptEditor

[https://tasmota.github.io/docs/Scripting-Language/#optional-external-editor](url)
